### PR TITLE
625 primary identifier

### PR DESCRIPTION
**Pull request by ahakanzn**

### CODE REVIEW

File: registry-persistence/src/main/resources/org/gbif/registry/persistence/mapper/collections/CollectionMapper.xml [hunk 1]
```diff
@@ -593,6 +594,28 @@
     WHERE regexp_replace(i.identifier, 'http://', '') = regexp_replace(#{identifier,jdbcType=VARCHAR}, 'http://', '')
       AND c.deleted IS NULL;
   </select>
+
+  <!-- Update existing primary identifier to false -->
+  <update id="updatePrimaryIdentifier">
```
Comment (marcos-lg) [inline registry-persistence/src/main/resources/org/gbif/registry/persistence/mapper/collections/CollectionMapper.xml:599]: maybe we can give it a more meaningful name? sth like setAllIdentifiersToNonPrimary or along those lines

File: registry-persistence/src/main/resources/org/gbif/registry/persistence/mapper/collections/CollectionMapper.xml [hunk 2]
```diff
@@ -593,6 +594,28 @@
...
+    SET is_primary=#{primary,jdbcType=BOOLEAN}
+    WHERE collection_key=#{targetEntityKey,jdbcType=OTHER} AND identifier_key=#{identifierKey,jdbcType=OTHER}
+  </update>
+
+  <select id="areRelated" resultType="boolean">
+    SELECT true
+    FROM collection_identifier
+    WHERE collection_key = #{targetEntityKey,jdbcType=OTHER}
+    AND identifier_key = #{identifierKey,jdbcType=INTEGER}
+    LIMIT 1
```
Comment (marcos-lg) [inline registry-persistence/src/main/resources/org/gbif/registry/persistence/mapper/collections/CollectionMapper.xml:617]: I'm not sure that we need the limit 1, I think it's more common to do select exists()

**Comment by marcos-lg** [comment]
Looks good to me overall. What I'm thinking is that so far we don't allow to update identifiers, they are immutable and with these changes we'd start to allow it. So what do you think about keeping them immutable as they are but you create a service to just set an identifier to primary or non-primary?

**Comment by ahakanzn** [comment]
> Looks good to me overall. What I'm thinking is that so far we don't allow to update identifiers, they are immutable and with these changes we'd start to allow it. So what do you think about keeping them immutable as they are but you create a service to just set an identifier to primary or non-primary?

Yes I also thought about that but couldn't be sure, that would be better since this we're doing this for only grscicoll entites

**Pull request merged**